scantool/scantool_cli.c

```diff
@@ -390,10 +390,12 @@ static int
 cmd_date(UNUSED(int argc), UNUSED(char **argv)) {
 	struct tm *tm;
 	time_t now;
+	char str[256];
 
 	now = time(NULL);
 	tm = localtime(&now);
-	printf("%s", asctime(tm));
+	strftime(str, sizeof(str), "%a %b %d %H:%M:%S %Y", tm);
+	printf("%s", str);
```

Comment:
Looks good, but C99 (7.23.3.5 *8) says if strftime can't fit in the given buffer : 

` zero is returned and the contents of the array are indeterminate.`
We should either initialize str with a safe dummy string, or handle the error somehow. It annoys me that there's no clean method to know for sure the required buffer size to guarantee strftime will succeed...